kong/db/schema/entities/ca_certificates.lua

```diff
@@ -8,10 +9,22 @@ return {
   fields = {
     { id = typedefs.uuid, },
     { created_at = typedefs.auto_timestamp_s },
-    { cert = typedefs.certificate { required = true, unique = true, }, },
+    { cert = typedefs.certificate { required = true }, },
+    { cert_digest = { type = "string", unique = true }, },
     { tags = typedefs.tags },
   },
 
+  check = function(entity)
+    local digest = str.to_hex(openssl_x509.new(entity.cert):digest("sha256"))
+    if not digest then
+      return nil, "cannot create digest value of certificate"
+    end
+
+    entity.cert_digest = digest
+
+    return true
+  end,
```

Comment:
I think the accepted way to do this kind of thing nowadays is via transformations. This might require changing the tests below:

```suggestion
  transformations = {
    {
      input = { "cert" },
      on_write = function(cert)
        local digest = str.to_hex(openssl_x509.new(cert):digest("sha256"))
        if not digest then
          return nil, "cannot create digest value of certificate"
        end
        return { digest = digest }
      end,
    },
  },
```

Reply:
Yes, if it is not with transformations it may cause issues with dbless.

Reply:
thanks @kikito and @bungle , working on it!